Okta/OIDAuthState+Okta.swift

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2019, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+import Hydra
+import OktaAppAuth
+
+// Okta Extension of OIDAuthState
+extension OIDAuthState {
+
+    static func getState(withAuthRequest authRequest: OIDAuthorizationRequest) -> Promise<OIDAuthState> {
+        return Promise<OIDAuthState>(in: .background, { resolve, reject, _ in
+            // setup custom URL session
+            self.setupURLSession()
+
+            // Make authCode request
+            OIDAuthorizationService.perform(authRequest: authRequest, callback: { authResponse, error in
+                guard let authResponse = authResponse else {
+                    return reject(OktaError.APIError("Authorization Error: \(error!.localizedDescription)"))
+                }
+
+                guard let _ = authResponse.authorizationCode,
+                      let tokenRequest = authResponse.tokenExchangeRequest() else {
+                        return reject(OktaError.unableToGetAuthCode)
+                }
+
+                // Make token request
+                OIDAuthorizationService.perform(tokenRequest, originalAuthorizationResponse: authResponse, callback:
+                { tokenResponse, error in
+                    guard let tokenResponse = tokenResponse else {
+                        return reject(OktaError.APIError("Authorization Error: \(error!.localizedDescription)"))
+                    }
+
+                    let authState = OIDAuthState(authorizationResponse: authResponse, tokenResponse: tokenResponse)
+                    return resolve(authState)
+                })
+            })
+        })
+        .always {
+            // Restore default URL session
+            self.restoreURLSession()
+        }
+    }
+    
+    private static func setupURLSession() {
+        /*
+         Setup auth session to block redirection because authorization request
+         implies redirection and passing authCode as a query parameter.
+        */
+        let config = URLSessionConfiguration.default
+        
+        // Fix issue when token request ends up with 403 error
+        config.httpAdditionalHeaders = ["Cookie" : "troute=t1;"]
```

Comment:
Was this required? I haven't seen a need for `troute=t1` before.

Reply:
Otherwise, tokens exchange request always fails with 403 status.

Reply:
@anastasiiaiurok-okta Can you reproduce this problem in Postman? The token endpoints should not require any `Cookie` header.

Reply:
The issue is not reproducible with Postman. And it is not reproducible when I am running auth request via browser (like we do for sign in and sign out).

Reply:
Probably for the GET requests server verifies "User-Agent" header field and rejects requests with unknown or empty User-Agent data. Just curious whether postman automatically adds User-Agent during request. 
@anastasiiaiurok-okta,  can we check with wireshark the difference in headers(sdk https header vs postman https header)?

Reply:
Comparing the headers haven't give any result. Adding a "User-Agent" header doesn't help as well. I think the problem is the same as the one described [here](https://devforum.okta.com/t/getting-status-code-403-when-trying-to-hit-access-token-endpoint/3809/3).

Reply:
@anastasiiaiurok-okta This is strange! Can you confirm that your code is **not** sending any other cookies if you remove `troute=1`?

Can you also dump a full Wireshark trace of both this SDK and Postman making the same request? I'd like to see if this issue can be replicated in curl (outside of this SDK).

Reply:
@nbarbettini 
I verified once more. It seams that some cookies are added when processing token request:
`	:method: POST
	:scheme: https
	:path: /oauth2/default/v1/token
	:authority: lohika-um.oktapreview.com
	cookie: JSESSIONID=F5F6BB5167BF20A4842F5CC084E27BB2; proximity_c2844254914385881afe1dc86fe02f83=WfUXcf4+kcbM4SddBkfVqoLasonFMGO3YZJsxXux4SdAWbQX9sOcbTaAQxapBTT6egbuzfUJV9LQFI5g7tQpsKB/6AoGP9kOq0LlvTdJuX1v4r9d3bKXHkQd+RHWnHRKYy8tOa6yqhpX1AmdwuQT4ZFv5aiXdqAoUE+qBjFBML0tNxUc+O5ZI2y+l+UymZ4q; sid=102BUOJupzRRlmlIiet8qJImQ; t=blue-dark; DT=DI07zFnRRqwT6KrZtBDxVWeDQ
	accept: */*
	content-type: application/x-www-form-urlencoded; charset=UTF-8
	user-agent: Okta_Example/1 CFNetwork/975.0.3 Darwin/18.2.0
	accept-language: en-us
	content-length: 202
	accept-encoding: br, gzip, deflate
`

Postman request looks a little bit different:
`POST /oauth2/default/v1/token HTTP/1.1
	Content-Type: application/x-www-form-urlencoded
	Accept: application/json
	User-Agent: PostmanRuntime/7.4.0
	Host: lohika-um.oktapreview.com
	cookie: JSESSIONID=9A00BCF3A8CC7D5F9A8952A472C746A3; t=blue-dark; DT=DI04QW_zKeaQB6NiUx71Rfufw
	accept-encoding: gzip, deflate
	content-length: 206
	Connection: close
`

I've disabled cockiest programmatically for auth requests. Looks like it helped resolve the issue.